src/PKSim.Core/Services/ParameterDefaultStateUpdater.cs

```diff
@@ -14,22 +15,44 @@ public class ParameterDefaultStateUpdater : IParameterDefaultStateUpdater
    {
       public void UpdateDefaultFor(ISpatialStructure spatialStructure)
       {
-         //TODO implement
+         var organism = spatialStructure.TopContainers.FindByName(Constants.ORGANISM);
+         if (organism == null) return;
+         updateDefaultStateForOriginDataParameters(organism);
       }
 
-
-      private void updateDefaultStateForOriginDataParameters(IContainer spatialStructureOrganism)
+      private void updateDefaultStateForOriginDataParameters(IContainer organism)
       {
-         new[] { CoreConstants.Parameter.AGE, Constants.Parameters.GESTATIONAL_AGE, CoreConstants.Parameter.HEIGHT }.Each(x =>
+         var parameterNames = new[]
          {
-            spatialStructureOrganism.Parameter(x).IsDefault = true;
-         });
-      }
+            CoreConstants.Parameter.AGE,
+            Constants.Parameters.GESTATIONAL_AGE,
+            CoreConstants.Parameter.HEIGHT
+         };
 
+         parameterNames
+            .Select(organism.Parameter)
+            .Where(p => p != null)
+            .Each(resetDefault);
+      }
 
       public void UpdateDefaultFor(IEventGroupBuildingBlock eventGroupBuildingBlock)
       {
-         //TODO implement
+         eventGroupBuildingBlock.Each(updateDefaultStateForEventParameters);
       }
+
+      private void updateDefaultStateForEventParameters(IEventGroupBuilder eventGroupBuilder)
+      {
+         var parameterNames = new[]
+         {
+            CoreConstants.Parameter.DOSE,
```

Comment:
These are all the event parameters that I could think of as well .. We might need more also